src/main/scala/viper/gobra/frontend/info/base/Type.scala

```diff
@@ -99,15 +99,29 @@ object Type {
     case object Send extends ChannelModus("chan<-")
   }
 
-  case class StructT(clauses: ListMap[String, (Boolean, Type)], decl: PStructType, context: ExternalTypeInfo) extends ContextualType {
-    lazy val fieldsAndEmbedded: ListMap[String, Type] = clauses.map(removeFieldIndicator)
-    lazy val fields: ListMap[String, Type] = clauses.filter(isField).map(removeFieldIndicator)
-    lazy val embedded: ListMap[String, Type] = clauses.filterNot(isField).map(removeFieldIndicator)
-    private def isField(clause: (String, (Boolean, Type))): Boolean = clause._2._1
-    private def removeFieldIndicator(clause: (String, (Boolean, Type))): (String, Type) = (clause._1, clause._2._2)
+  trait StructClauseT extends Type {
```

Comment:
why are we making this a subtype of `Type`? We are adding a case that we have to dismiss every time we pattern match on `Type`, which sounds unnecessary.